### ISSUE TITLE
Regression when upgrading from 3.12.2 to 3.32.2

### ISSUE THREAD

**Issue by Blitz2145** [pulumi/pulumi-aws#1402]
After upgrading to 3.32.2 of the pulumi/aws package, we are getting errors from the provider in fetching the credentials provided by the instance role profile.

## Expected behavior
pulumi preview should succeed and show no changes.

## Current behavior
pulumi preview fails with credentials error

## Steps to reproduce

1. Run any pulumi typescript code with `pulumi/aws@3.12.2`. `pulumi preview` command works without credentials errors.
2. Upgrade to `pulumi/aws@3.32.2`
3. `pulumi preview` fails

## Logs

Traceback
```
    error: Running program '/home/ubuntu/src/blah/deployment' failed with an unhandled exception:
    Error: invocation of aws:sns/getTopic:getTopic returned an error: 1 error occurred:
        * error configuring Terraform AWS Provider: no valid credential sources for Terraform AWS Provider found.
    Please see https://registry.terraform.io/providers/hashicorp/aws
    for more information about providing credentials.
    Error: NoCredentialProviders: no valid providers in chain. Deprecated.
        For verbose messaging see aws.Config.CredentialsChainVerboseErrors
        at Object.callback (/home/ubuntu/src/blah/deployment/node_modules/@pulumi/pulumi/runtime/invoke.js:139:33)
        at Object.onReceiveStatus (/home/ubuntu/src/blah/deployment/node_modules/@grpc/grpc-js/src/client.ts:334:26)
        at Object.onReceiveStatus (/home/ubuntu/src/blah/deployment/node_modules/@grpc/grpc-js/src/client-interceptors.ts:426:34)
        at Object.onReceiveStatus (/home/ubuntu/src/blah/deployment/node_modules/@grpc/grpc-js/src/client-interceptors.ts:389:48)
        at /home/ubuntu/src/blah/deployment/node_modules/@grpc/grpc-js/src/call-stream.ts:276:24
        at processTicksAndRejections (internal/process/task_queues.js:75:11)
    Error: invocation of aws:sns/getTopic:getTopic returned an error: 1 error occurred:
        * error configuring Terraform AWS Provider: no valid credential sources for Terraform AWS Provider found.
    Please see https://registry.terraform.io/providers/hashicorp/aws
    for more information about providing credentials.
    Error: NoCredentialProviders: no valid providers in chain. Deprecated.
        For verbose messaging see aws.Config.CredentialsChainVerboseErrors
        at Object.callback (/home/ubuntu/src/blah/deployment/node_modules/@pulumi/pulumi/runtime/invoke.js:139:33)
        at Object.onReceiveStatus (/home/ubuntu/src/blah/deployment/node_modules/@grpc/grpc-js/src/client.ts:334:26)
        at Object.onReceiveStatus (/home/ubuntu/src/blah/deployment/node_modules/@grpc/grpc-js/src/client-interceptors.ts:426:34)
        at Object.onReceiveStatus (/home/ubuntu/src/blah/deployment/node_modules/@grpc/grpc-js/src/client-interceptors.ts:389:48)
        at /home/ubuntu/src/blah/deployment/node_modules/@grpc/grpc-js/src/call-stream.ts:276:24
        at processTicksAndRejections (internal/process/task_queues.js:75:11)
    Error: invocation of aws:sns/getTopic:getTopic returned an error: 1 error occurred:
        * error configuring Terraform AWS Provider: no valid credential sources for Terraform AWS Provider found.
    Please see https://registry.terraform.io/providers/hashicorp/aws
    for more information about providing credentials.
    Error: NoCredentialProviders: no valid providers in chain. Deprecated.
        For verbose messaging see aws.Config.CredentialsChainVerboseErrors
        at Object.callback (/home/ubuntu/src/blah/deployment/node_modules/@pulumi/pulumi/runtime/invoke.js:139:33)
        at Object.onReceiveStatus (/home/ubuntu/src/blah/deployment/node_modules/@grpc/grpc-js/src/client.ts:334:26)
        at Object.onReceiveStatus (/home/ubuntu/src/blah/deployment/node_modules/@grpc/grpc-js/src/client-interceptors.ts:426:34)
        at Object.onReceiveStatus (/home/ubuntu/src/blah/deployment/node_modules/@grpc/grpc-js/src/client-interceptors.ts:389:48)
        at /home/ubuntu/src/blah/deployment/node_modules/@grpc/grpc-js/src/call-stream.ts:276:24
        at processTicksAndRejections (internal/process/task_queues.js:75:11)
```

## Context (Environment)
Prevents deployments on specific instance.
OS & Versions
```
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.3 LTS
Release:	18.04
Codename:	bionic
node: v14.15.1
yarn: 1.22.5
pulumi cli: v2.22.0
```

Lock File (Before)
```
"@pulumi/aws@^3.8.0":
  version "3.12.2"
  resolved "https://registry.yarnpkg.com/@pulumi/aws/-/aws-3.12.2.tgz#9686a7274d67f6e1d2f53f3462a9718e9433cae5"
  integrity sha512-azoH3r2bI3KgK5Yyu2F2wL3yd+d8pnEblPmpPinSiJ8rgyv87E7Ot9GjEQgniuHaB1MODzox3B8ySflqwY9akA==
  dependencies:
    "@pulumi/pulumi" "^2.0.0"
    aws-sdk "^2.0.0"
    builtin-modules "3.0.0"
    mime "^2.0.0"
    read-package-tree "^5.2.1"
    resolve "^1.7.1"
```

Lock File (After)
```
"@pulumi/aws@^3.32.2":
  version "3.32.2"
  resolved "https://registry.yarnpkg.com/@pulumi/aws/-/aws-3.32.2.tgz#b4d6ef49c591c2b40ffb783724cdc2df3f077d29"
  integrity sha512-gc41+qYDdTP5OULAx8igRwn5a9GLmeW546ORfkH/M1K0ed6BEOKavnbslS1U8noVLMv+cxMGlcp9Oj6SfE1WHA==
  dependencies:
    "@pulumi/pulumi" "^2.17.0"
    aws-sdk "^2.0.0"
    builtin-modules "3.0.0"
    mime "^2.0.0"
    read-package-tree "^5.2.1"
    resolve "^1.7.1"
```

Plugin LS output
```
NAME        KIND      VERSION  SIZE    INSTALLED  LAST USED
aws         resource  3.32.2   249 MB  n/a        39 minutes ago
```

I can try to bisect further if needed, let me know.

**Comment by stack72** [comment]
Hi @Blitz2145 

There was a long issue about this until the weekend just past - I am going to be adding a section to our README.

TL;DR we changed the provider configuration defaults to remove redundant calls to AWS in a bid to speed up our provider

This means, that when using pulumi-aws under an IAM Role to get credentials, you would need to do 1 of the following 2 things:

1. When using the default provider (no instances of aws.Provider)
* pulumi config set aws:skipMetadataApiCheck false

2. If using a named provider, then set skipMetadataApiCheck as part of the provider block:
```
const prov = new aws.Provider("named-provider", {
    ...
    skipMetadataApiCheck: false,
});
```

I apologise this caused you pain but I literally released a fix that allowed this to work on Sunday

I will get the README of the project updated tomorrow

Paul

**Comment by Blitz2145** [comment]
Awesome, I'll try that out.

**Comment by Blitz2145** [comment]
Confirmed:

```
pulumi config set aws:skipMetadataApiCheck false
```

This config change fixes the issue for those relying on the credentials from instance profile. Thanks!

**Comment by stack72** [comment]
Ok, I *promise* I will get our README updated tomorrow!!
